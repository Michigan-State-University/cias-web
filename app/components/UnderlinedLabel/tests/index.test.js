import React from 'react';
import { render } from '@testing-library/react';
import 'jest-styled-components';

import { UnderlinedLabel } from '../index';

describe('<UnderlinedLabel />', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    render(<UnderlinedLabel text="Test Label" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('should render and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<UnderlinedLabel text="Test Label" />);
    expect(firstChild).toMatchSnapshot();
  });

  it('should render isActive and match the snapshot', () => {
    const {
      container: { firstChild },
    } = render(<UnderlinedLabel text="Test Label" isActive />);

    expect(firstChild).toMatchSnapshot();
  });
});
