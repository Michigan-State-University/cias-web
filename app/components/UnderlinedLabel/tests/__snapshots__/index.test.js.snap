// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UnderlinedLabel /> should render and match the snapshot 1`] = `
.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  position: relative;
  bottom: -3px;
  margin: 0 10px;
  padding-bottom: 7px;
  border-bottom-width: 2px;
  border-bottom-color: #B727EA;
  border-bottom-style: none;
  cursor: pointer;
  color: #9AA1B6 !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #2F3850;
}

<div
  class="c0"
>
  Test Label
</div>
`;

exports[`<UnderlinedLabel /> should render isActive and match the snapshot 1`] = `
.c0 {
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  height: -webkit-max-content;
  height: -moz-max-content;
  height: max-content;
  position: relative;
  bottom: -3px;
  margin: 0 10px;
  padding-bottom: 7px;
  border-bottom-width: 2px;
  border-bottom-color: #B727EA;
  border-bottom-style: solid;
  cursor: pointer;
  color: #2F3850 !important;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c0:hover {
  color: #2F3850;
}

<div
  class="c0"
>
  Test Label
</div>
`;
